easel/app.js

```diff
@@ -88,28 +87,34 @@ var executor = function (args, success, failure) {
   if (!args.params['DXF File']) {
     return failure("Please upload a DXF file on the left to begin.");
   }
-  var request = new AWS.Lambda({
-    region: LAMBDA_REGION,
-    accessKeyId: AWS_ACCESS_KEY,
-    secretAccessKey: SECRET_ACCESS_KEY,
-    maxRetries: 3
-  }).invoke({
-    FunctionName: LAMBDA_NAME,
-    Payload: JSON.stringify({ 'url': args.params['DXF File'] })
-
-  }, function (err, result) {
-    if (err) {
-      failure("DXF import failed: " + err + ". Try re-uploading the file or refreshing your browser.");
-    } else {
-      var payload = JSON.parse(result.Payload);
-      if (payload.errorMessage || !payload.svg) {
-        var errorMessage = payload.errorMessage;
-        if (errorMessage.match('delegate')) {
-          errorMessage = "Cannot read file";
-        }
-        failure("DXF import failed: " + errorMessage);
-      } else {
-        var svg = payload.svg;
+
+  fetch(apiGatewayName, {
+    method: "POST",
+    body: JSON.stringify({ url: args.params["DXF File"] }),
+    headers: {
+      "x-api-key": apiKey,
+      "Content-Type": "text/plain",
+    }
+  })
+    .then(response => response.json())
+    .then(function(response) {
+      /**
+       * The lambda app finshed! result contains the apps payload, parsed as JSON
+       * The example app returns an svg as the only item in the map
+       *
+       * Since this is a "v1" app, we return SVG data
+       */
+      result = (typeof(response) == "string") ? JSON.parse(response) : response;
+
+      /**
+       * Process errors from inside of the Lambda app if returned payload has any
+       */
+      if ("errorMessage" in result) {
+        failure(result.errorMessage);
+      }
+
+      if ("svg" in result) {
```

Comment:
do we want to warn or log something if there's no svg?

Reply:
Addressed in https://github.com/inventables/dxfimport/pull/3/commits/2fd41a0fb46a4382557f683a55e9a09b551707de